src/type_sys/fender_reference/mod.rs

```diff
@@ -139,8 +139,10 @@ impl Value for FenderReference {
         }
     }
 
-    fn gen_list(values: Vec<Self>) -> Self {
-        FenderReference::FRaw(FenderValue::List(InternalReference::new(values)))
+    fn gen_list(values: impl IntoIterator<Item = Self>) -> Self {
+        FenderReference::FRaw(FenderValue::List(InternalReference::new(
+            values.into_iter().collect(),
```

Comment:
What is the point of this? converting to an iterator and then back to a vec would be slower and if its ever needed it could be done on the callers side

Reply:
it's not coming from a Vec necessarily, everything is passing args with more lightweight types where possible now. you can pass arguments using a stack-allocated array now, for example.

Reply:
but its still getting converted to a vec 
so this changes nothing if somthing other than a vec is passed (other than being neater), but if a vec is passed then it will slow things down

also the only time I can see this being used is in `execution_engine.rs:125` and I think its passing a vec there

Reply:
we will also want to be changing how we pass arguments in our stdlib functions to use arrays and not vecs. the *main* situation where this will be getting used is going to be when args are passed through Freight internally via the expressions, and this is done without vecs. there's a vec that holds the expression form of the arguments, but this has to be converted into the actual type system's value type. this was being done with a vec before and now uses an iterator, which eliminates overhead.